Detectors/Upgrades/ALICE3/FCT/simulation/src/Detector.cxx

```diff
@@ -0,0 +1,592 @@
+
+  LOG(info) << "Exporting FCT Detector layout to " << configFileName;
+
+  std::ofstream fOut(configFileName, std::ios::out);
+  if (!fOut) {
+    printf("Cannot open file\n");
+    return;
+  }
+  fOut << "#   z_layer   r_in   r_out_l_side   Layerx2X0" << std::endl;
+  for (auto layer : mBackwardLayers) {
+    if (layer.getType() == 0) {
+      fOut << layer.getZ() << "  " << layer.getInnerRadius() << "  " << layer.getOuterRadius() << "  " << layer.getx2X0() << std::endl;
+    } else if (layer.getType() == 1) {
+      fOut << layer.getZ() << "  " << layer.getInnerRadius() << "  " << layer.getSideLength() << "  " << layer.getx2X0() << std::endl;
+    }
+  }
+}
+
+//_________________________________________________________________________________________________
+void Detector::buildBasicFCT(const FCTBaseParam& param)
+{
+  // Build a basic parametrized FCT detector with nLayers equally spaced between z_first and z_first+z_length
+  // Covering pseudo rapidity [etaIn,etaOut]. Silicon thinkness computed to match layer x/X0
+
+  LOG(info) << "Building FCT Detector: Conical Telescope";
+
+  auto z_first = param.z0;
+  auto z_length = param.zLength;
+  auto etaIn = param.etaIn;
+  auto etaOut = param.etaOut;
+  auto Layerx2X0 = param.Layerx2X0;
+  mNumberOfLayers = param.nLayers;
+  mLayerID.clear();
+
+  Int_t type = 0; // Disk
+
+  for (Int_t layerNumber = 0; layerNumber < mNumberOfLayers; layerNumber++) {
+    std::string layerName = GeometryTGeo::getFCTLayerPattern() + std::to_string(layerNumber); // + mNumberOfLayers * direction);
+    mLayerName.push_back(layerName);
+
+    // Adds evenly spaced layers
+    Float_t layerZ = z_first + (layerNumber * z_length / (mNumberOfLayers - 1)) * std::copysign(1, z_first);
+    Float_t rIn = std::abs(layerZ * std::tan(2.f * std::atan(std::exp(-etaIn))));
+    Float_t rOut = std::abs(layerZ * std::tan(2.f * std::atan(std::exp(-etaOut))));
+    auto& thisLayer = mBackwardLayers.emplace_back(0, layerNumber, layerName, layerZ, rIn, rOut, Layerx2X0, type);
+  }
+}
+
+//_________________________________________________________________________________________________
+void Detector::buildFCTV1()
+{
+  // Build default FCT detector
+
+  LOG(info) << "Building FCT Detector: V1";
+
+  mNumberOfLayers = 9;
+  Float_t layersx2X0 = 1.e-2;
+
+  std::vector<std::array<Float_t, 4>> layersConfig{
+    {-442.0, 6.0, 44.1, layersx2X0}, // {z_layer, r_in, r_out, Layerx2X0}
+    {-444.0, 6.0, 44.3, layersx2X0},
+    {-446.0, 6.0, 44.5, layersx2X0},
+    {-448.0, 6.0, 44.7, layersx2X0},
+    {-450.0, 6.1, 44.9, layersx2X0},
+    {-460.0, 6.2, 45.9, layersx2X0},
+    {-470.0, 6.3, 46.9, layersx2X0},
+    {-480.0, 6.5, 47.9, layersx2X0},
+    {-490.0, 6.6, 48.9, layersx2X0}};
+
+  mLayerID.clear();
+  mLayerName.clear();
+  mBackwardLayers.clear();
+
+  Int_t type = 0; // Disk
+
+  for (int layerNumber = 0; layerNumber < mNumberOfLayers; layerNumber++) {
+    std::string directionName = std::to_string(0);
```

Comment:
why are you defining this variable? Is it supposed to be actually used? Right now it looks like you could just add a "0" at the next line where you use it

Reply:
The FT3, the code the FCT was based on, has backward layers, "0", and forward layers, "1". When I wanted to get rid of this a conflict occurred in finding the layers when they were later called by name. This is hopefully resolved as of now since I moved the initialization statements in the Detector::Detector class to Detector::ConstructGeometry. As I learned, a dummy detector object is created wehn the detector is called. Thus I moved this to the ConstructGeometry class. This is not done yet in the FT3 and a pull request should be created to fix that there too

Reply:
Resolved. This has its own commit: https://github.com/AliceO2Group/AliceO2/pull/8679/commits/ccdaf8c36ad4cc11041ce94bd33727d616766e0f

Reply:
I see, thanks!
Just as an advice maybe: A line like this - regardless of whether this code was adapted from other code - looks weird if there is nothing else done other than just making a string from a number